lib/provider.js

```diff
@@ -25,21 +26,133 @@ module.exports = function (dependencies) {
       recipients = [recipients];
     }
 
-    return Promise.all(recipients.map(token => this.client.write(builtNotification, token))).then(
-      responses => {
-        const sent = [];
-        const failed = [];
-
-        responses.forEach(response => {
-          if (response.status || response.error) {
-            failed.push(response);
-          } else {
-            sent.push(response);
-          }
-        });
-        return { sent, failed };
+    const sentNotifications = await Promise.allSettled(
```

Comment:
Here's a change that should result in improve expected results to developers (I think there are some GitHub issues related to this, I'll find them later). Basically using `allSettled` won't break if there is an individual notification error and the method will provide results for all attempted notifications whether they fail or succeed.